graphistry/embed_utils.py

```diff
@@ -0,0 +1,500 @@
+from matplotlib.font_manager import X11FontDirectories
+import numpy as np
+import torch
+import torch.nn as nn
+from torch.utils.data import Dataset, DataLoader
+from tqdm import trange
+from .networks import RGCNEmbed
+import dgl
+from dgl.dataloading import GraphDataLoader
+import torch.nn.functional as F
+
+import pandas as pd
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+class EmbedDistScore:
+
+    @staticmethod
+    def TransE(h, r, t):
+        return (h + r - t).norm(p=1, dim=1)
+
+    @staticmethod
+    def DistMult(h, r, t):
+        return (h * r * t).sum(dim=1)
+
+    @staticmethod
+    def RotatE(h, r, t):
+        return -(h * r - t).norm(p=1, dim=1)
+ 
+    
+
+class HeterographEmbedModuleMixin(nn.Module):
```

Comment:
nn.Module probably needs to be lazy somehow too

maybe there's a way to fully defer this file being imported, and that way we can keep top-level?